src/Messenger/IndexationRequestHandler.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace JoliCode\Elastically\Messenger;
+
+use Elastica\Document;
+use JoliCode\Elastically\Client;
+use Psr\Log\NullLogger;
+use Symfony\Component\Messenger\Exception\UnrecoverableMessageHandlingException;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+abstract class IndexationRequestHandler implements MessageHandlerInterface
+{
+    const OP_INDEX = 'index';
+    const OP_DELETE = 'delete';
+    const OP_UPDATE = 'update';
+    const OP_CREATE = 'create';
+
```

Comment:
There is no support for bulk?

Reply:
Not at the moment. I'm not sure how best to do it - didn't need such support in other projects.

Reply:
In a project I work on, we're introducing bulking operations in a messenger consumer for elasticsearch sync.

Not yet finished, but the gain is around 65% in terms of time.

If we adopt elastically (my planing) it would be great not get blocked in this subject. I mean I'm for "non flush" + "bulk"

Reply:
For "bulk"; it means being able to apply the same operation on multiple ID's. That's a good suggestion I will change the Messenger message to support this.

But I don't get the "non flush" part: if I don't flush the Indexer, updates will be lost. @lyrixx made the same suggestion, I have difficulties understanding how it can work. Do you suggest being able to run:

```php
$this->bus->dispatch(new IndexationRequest(....));
$this->indexer->flush();
```

??

Reply:
as stated in another comment, there are no issue with the flush (since the handler can be used in a async transport)

Reply:
Support for Bulk is added!